internal/recorder/http_recorder.go

```diff
@@ -0,0 +1,137 @@
+package recorder
+
+import (
+	"encoding/json"
+	"net/http"
+	"os"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/auth0/go-auth0"
+	"github.com/auth0/go-auth0/management"
+	"github.com/stretchr/testify/require"
+	"gopkg.in/dnaeon/go-vcr.v3/cassette"
+	"gopkg.in/dnaeon/go-vcr.v3/recorder"
+)
+
+const (
+	recordingsDIR    = "./../../test/data/recordings/"
+	recordingsTenant = "terraform-provider-auth0-dev"
+
+	// RecordingsDomain is used for testing with our recorded http interactions.
+	RecordingsDomain = recordingsTenant + ".eu.auth0.com"
+)
+
+// Recorder used for our http test mocks.
+type Recorder struct {
+	*recorder.Recorder
+}
+
+// New creates a new instance of our http recorder used in tests.
+func New(t *testing.T) *Recorder {
+	t.Helper()
+
+	httpRecordings := os.Getenv("AUTH0_HTTP_RECORDINGS")
+	httpRecordingsEnabled := httpRecordings == "true" || httpRecordings == "1" || httpRecordings == "on"
+	if !httpRecordingsEnabled {
+		return nil
+	}
+
+	recorderTransport, err := recorder.NewWithOptions(
+		&recorder.Options{
+			CassetteName:       recordingsDIR + t.Name(),
+			Mode:               recorder.ModeRecordOnce,
+			SkipRequestLatency: true,
+		},
+	)
+	require.NoError(t, err)
+
+	removeSensitiveDataFromRecordings(t, recorderTransport)
+
+	t.Cleanup(func() {
+		err := recorderTransport.Stop()
+		require.NoError(t, err)
+	})
+
+	return &Recorder{recorderTransport}
+}
+
+func removeSensitiveDataFromRecordings(t *testing.T, recorderTransport *recorder.Recorder) {
+	requestHeaders := []string{"Authorization"}
+	responseHeaders := []string{
+		"Alt-Svc",
+		"Cache-Control",
+		"Cf-Cache-Status",
+		"Cf-Ray",
+		"Date",
+		"Expect-Ct",
+		"Link",
+		"Ot-Baggage-Auth0",
+		"Ot-Baggage-Auth0-Request-Id",
+		"Ot-Tracer-Sampled",
+		"Ot-Tracer-Spanid",
+		"Ot-Tracer-Traceid",
+		"Server",
+		"Set-Cookie",
+		"Strict-Transport-Security",
+		"Traceparent",
+		"Tracestate",
+		"Vary",
+		"X-Content-Type-Options",
+		"X-Ratelimit-Limit",
+		"X-Ratelimit-Remaining",
+		"X-Ratelimit-Reset",
+	}
+
+	recorderTransport.AddHook(
+		func(i *cassette.Interaction) error {
+			for _, header := range requestHeaders {
+				delete(i.Request.Headers, header)
+			}
+			for _, header := range responseHeaders {
+				delete(i.Response.Headers, header)
```

Comment:
Do we need any particular header, or can all be removed?

Reply:
I think just the content-type and the user agent are useful for us.

Reply:
Maybe we could go with an allow-list of just those?

Reply:
[ce38495](https://github.com/auth0/terraform-provider-auth0/pull/309/commits/ce38495bf528e832062b7a9ef6f9e19a6137b0f5) Updated @Widcket 🎉